components/brave_wallet_ui/page/async/wallet_page_async_handler.ts

```diff
@@ -173,36 +213,23 @@ handler.on(WalletPageActions.checkWalletsToImport.getType(), async (store) => {
       BraveWallet.ExternalWalletType.MetaMask)
   store.dispatch(WalletPageActions.setCryptoWalletsInitialized(cwResult.initialized))
   store.dispatch(WalletPageActions.setMetaMaskInitialized(mmResult.initialized))
+  store.dispatch(WalletPageActions.setImportWalletsCheckComplete(true))
 })
 
 handler.on(WalletPageActions.importFromCryptoWallets.getType(), async (store: Store, payload: ImportFromExternalWalletPayloadType) => {
-  const braveWalletService = getWalletPageApiProxy().braveWalletService
-  const keyringService = getWalletPageApiProxy().keyringService
-  const result =
-    await braveWalletService.importFromExternalWallet(
-      BraveWallet.ExternalWalletType.CryptoWallets, payload.password, payload.newPassword)
-  if (result.success) {
-    keyringService.notifyWalletBackupComplete()
-  }
-  store.dispatch(WalletPageActions.setImportWalletError({
-    hasError: !result.success,
-    errorMessage: result.errorMessage ?? undefined
-  }))
+  const results: ImportWalletErrorPayloadType = await importFromExternalWallet(
+    BraveWallet.ExternalWalletType.CryptoWallets,
+    payload
+  )
+  store.dispatch(WalletPageActions.setImportWalletError(results))
 })
 
 handler.on(WalletPageActions.importFromMetaMask.getType(), async (store: Store, payload: ImportFromExternalWalletPayloadType) => {
-  const braveWalletService = getWalletPageApiProxy().braveWalletService
-  const keyringService = getWalletPageApiProxy().keyringService
-  const result =
-    await braveWalletService.importFromExternalWallet(
-      BraveWallet.ExternalWalletType.MetaMask, payload.password, payload.newPassword)
-  if (result.success) {
-    keyringService.notifyWalletBackupComplete()
-  }
-  store.dispatch(WalletPageActions.setImportWalletError({
-    hasError: !result.success,
-    errorMessage: result.errorMessage ?? undefined
-  }))
+  const results: ImportWalletErrorPayloadType = await importFromExternalWallet(
+    BraveWallet.ExternalWalletType.MetaMask,
+    payload
+  )
+  store.dispatch(WalletPageActions.setImportWalletError(results))
```

Comment:
Updated wallet import implementation so that we can check the import password before completing the import process